### ISSUE TITLE
Out of Memory Error while Streaming

### ISSUE THREAD

**Issue by dob** [livepeer/go-livepeer#181]
Captured by @rayj00. Streaming started fine, but after segment 236, the node crashed with an out of memory error:

```
1113 16:26:34.344611 18028 player.go:90] LPMS got HTTP request @ /stream/1220cdb4efa44f9757e6690fae183092fde30c401a150050578bb5102bd9 a423de49dce361c57d5a6fa801833b4bcc16ede4e87b8ceabe5b53682b760473c866d2ed_236.ts
fatal error: runtime: out of memory
```

**Comment by ericxtang** [comment]
I think this is because we keep the video in memory until it's finished.  Should probably re-visit this at some point.

**Comment by rayj00** [comment]
What memory are you talking about?  Ram or disk?

**Comment by ericxtang** [comment]
fixed by #255
